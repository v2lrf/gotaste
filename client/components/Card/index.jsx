import React from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'

function Card({ title, textCenter, children }) {
  const cardClasses = classnames('bg-white rounded shadow-lg', {
    'text-center': textCenter
  })

  return (
    <div className={cardClasses}>
      <div className="flex flex-row items-center">
        <div className="flex-1">
          {title && <h4 className="p-4 uppercase text-red-light">{title}</h4>}
          <div className="px-4 pb-4">{children}</div>
        </div>
      </div>
    </div>
  )
}

Card.propTypes = {
  title: PropTypes.string,
  children: PropTypes.node.isRequired,
  textCenter: PropTypes.bool
}

Card.defaultProps = {
  title: null,
  textCenter: false
}

export default Card
