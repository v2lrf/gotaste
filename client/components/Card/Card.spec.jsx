import React from 'react'
import Card from '.'

describe('Card', () => {
  let component
  beforeEach(() => {
    component = shallow(<Card>Some content!</Card>)
  })

  describe('Snapshots', () => {
    it('should match the default snapshot', () => {
      expect(component).toMatchSnapshot()
    })

    it('should match a snapshot with a title', () => {
      component.setProps({ title: 'Some title' })
      expect(component).toMatchSnapshot()
    })
  })
})
