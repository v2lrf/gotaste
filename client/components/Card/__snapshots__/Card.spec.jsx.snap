// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Snapshots should match a snapshot with a title 1`] = `
<div
  className="bg-white rounded shadow-lg"
>
  <div
    className="flex flex-row items-center"
  >
    <div
      className="flex-1"
    >
      <h4
        className="p-4 uppercase text-red-light"
      >
        Some title
      </h4>
      <div
        className="px-4 pb-4"
      >
        Some content!
      </div>
    </div>
  </div>
</div>
`;

exports[`Card Snapshots should match the default snapshot 1`] = `
<div
  className="bg-white rounded shadow-lg"
>
  <div
    className="flex flex-row items-center"
  >
    <div
      className="flex-1"
    >
      <div
        className="px-4 pb-4"
      >
        Some content!
      </div>
    </div>
  </div>
</div>
`;
